managed/CounterStrikeSharp.API/Modules/Menu/CenterHtmlMenu.cs

```diff
@@ -0,0 +1,125 @@
+﻿/*
+ *  This file is part of CounterStrikeSharp.
+ *  CounterStrikeSharp is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  CounterStrikeSharp is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with CounterStrikeSharp.  If not, see <https://www.gnu.org/licenses/>. *
+ */
+
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace CounterStrikeSharp.API.Modules.Menu
+{
+    public class CenterHtmlMenu : BaseMenu
+    {
+        public CenterHtmlMenu(string title) : base(ModifyTitle(title))
+        {
+        }
+        
+        public override ChatMenuOption AddMenuOption(string display, Action<CCSPlayerController, ChatMenuOption> onSelect, bool disabled = false)
+        {
+            var option = new ChatMenuOption(ModifyOptionDisplay(display), disabled, onSelect);
+            MenuOptions.Add(option);
+            return option;
+        }
+        
+        private static string ModifyTitle(string title)
+        {
+            if (title.Length > 32)
+            {
+                Application.Instance.Logger.LogWarning("Title should not be longer than 32 characters for a CenterHtmlMenu");
+                return title[..32];
+            }
+
+            return title;
+        }
+
+        private static string ModifyOptionDisplay(string display)
+        {
+            if (display.Length > 26)
+            {
+                Application.Instance.Logger.LogWarning("Display should not be longer than 26 characters for a CenterHtmlMenu item");
+                return display[..26];
+            }
+
+            return display;
+        }
+    }
+
+    public class CenterHtmlMenuInstance : BaseMenuInstance
+    {
+        private readonly BasePlugin _plugin;
+
+        public CenterHtmlMenuInstance(BasePlugin plugin, CCSPlayerController player, IMenu menu) : base(player, menu)
+        {
+            _plugin = plugin;
+            RemoveOnTickListener();
```

Comment:
Would this not cause issues with plugins that already have `OnTick` defined?

Reply:
Yes good catch, as it calls `_plugin.RemoveListener("OnTick", onTick);` the whole plugin would stop receiving `OnTick`

Reply:
I don’t think it clashes, the listener has a key and this function just removes the display function from the array of OnTick listeners afaik. I can test it to make sure the other OnTick listeners aren’t cleared but I’m pretty sure they won’t be.

Reply:
Yeah it should be using the actual delegate passed through to remove; so it can only remove listeners it has the actual underlying method being called, so this shouldn't cause any issues